### DISCUSSION TITLE
completion_waiting_dots

### DISCUSSION THREAD

**Discussion by realmattneale** [ohmyzsh/ohmyzsh#10785] [Help] [answered]
So I recently got oh-my-zsh installed and configured the terminal with plugins that I wanted and they all work fine. The only thing that isn't working for me is the completion_waiting_dots feature. I have mine set to "true" but I never see them pop up. I tried copying a large directory that would take awhile to finish and the cursor just moves to the next line and blinks like normal. I've tried changing the theme and removing plugins but nothing has worked yet.

I was also wondering where is the actual code for the completion_waiting_dots because maybe I could put it in a separate file and source that to test it further. Thanks.

**Comment by carlosala** [comment, marked as answer]
You have the code [here](https://github.com/ohmyzsh/ohmyzsh/blob/c96fc233c4903ba75fba5bc55e7f91f9dc8e460e/lib/completion.zsh#L61-L75}. By the way, it's really weird that this doesn't work for you. Maybe define an infinite loop in some command completion and test if appears. Common completion funcions are really fast and it's difficult to see if appears.

**Reply by realmattneale** [reply]
Thanks for directing me to that code. Yeah it's strange because I have zsh 5.8 and just updated oh-my-zsh and not using a multi-line prompt either. I thought the dots were supposed to show no matter what as long as the terminal is waiting for a command to finish. So like even a basic "sleep 10" should still show them right?  If I cp a large directory I thought they should print out until that is finished. I'm on WSL2 with Ubuntu and Windows Terminal but I wouldn't think any of that would make a difference.

**Reply by carlosala** [reply]
This is only for completion. So the time between you press the tab and the completion appears.
